tests/xtro-sharpie/SimdCheck.cs

```diff
@@ -0,0 +1,382 @@
+					// We can't map this method to a native function.
+					Console.WriteLine ($"!simd-can't-map-managed! {method}");
+				}
+				return;
+			}
+
+			ManagedSimdInfo existing;
+			if (managed_methods.TryGetValue (key, out existing)) {
+				if (very_strict)
+					Console.WriteLine ($"!simd-double-mapping! same key '{key}' for both '{existing.Method}' and '{method}'");
+			} else {
+				managed_methods [key] = new ManagedSimdInfo {
+					Method = method, ContainsInvalidMappingForSimd = invalid_simd_type
+				};
+			}
+		}
+
+		bool ContainsSimdTypes (MethodDefinition method, ref bool invalid_for_simd)
+		{
+			if (IsSimdType (method.ReturnType, ref invalid_for_simd))
+				return true;
+
+			if (method.HasParameters) {
+				foreach (var param in method.Parameters)
+					if (IsSimdType (param.ParameterType, ref invalid_for_simd))
+						return true;
+			}
+
+			return false;
+		}
+
+		bool IsSimdType (TypeReference td, ref bool invalid_for_simd)
+		{
+			return managed_simd_types.TryGetValue (td.Name, out invalid_for_simd);
+		}
+
+		bool ContainsSimdTypes (ObjCMethodDecl decl, ref string simd_type, ref bool requires_marshal_directive)
+		{
+			if (IsSimdType (decl.ReturnQualType, ref simd_type, ref requires_marshal_directive))
+				return true;
+
+			var is_simd_type = false;
+			foreach (var param in decl.Parameters)
+				is_simd_type |= IsSimdType (param.QualType, ref simd_type, ref requires_marshal_directive);
+
+			return is_simd_type;
+		}
+
+		bool IsExtVector (QualType type, ref string simd_type)
+		{
+			var rv = false;
+			var t = type.CanonicalQualType.Type;
+
+			// Unpoint the type
+			var pointerType = t as Clang.Ast.PointerType;
+			if (pointerType != null)
+				t = pointerType.PointeeQualType.Type;
+
+			if (t.Kind == TypeKind.ExtVector) {
+				rv = true;
+			} else {
+				var r = (t as RecordType)?.Decl;
+				if (r != null) {
+					foreach (var f in r.Fields) {
+						var qt = f.QualType.CanonicalQualType.Type;
+						if (qt.Kind == TypeKind.ExtVector) {
+							rv = true;
+							break;
+						}
+						var at = qt as ConstantArrayType;
+						if (at != null) {
+							if (at.ElementType.Type.Kind == TypeKind.ExtVector) {
+								rv = true;
+								break;
+							}
+						}
+					}
+				}
+			}
+
+			var typeName = type.ToString ();
+
+			if (!rv && typeName.Contains ("simd"))
+				Console.WriteLine ($"!simd-unknown-type! Could not detect that {typeName} is a Simd type, but its name contains 'simd'.");
+			
+			if (rv)
+				simd_type = typeName;
+			
+			return rv;
+		}
+
+		bool IsSimdType (QualType type, ref string simd_type, ref bool requires_marshal_directive)
+		{
+			var str = Undecorate (type.ToString ());
+
+			if (type_mapping.TryGetValue (str, out var info)) {
+				requires_marshal_directive = true;
+				simd_type = str;
+				return true;
+			}
+
+			if (IsExtVector (type, ref simd_type))
+				Console.WriteLine ($"!simd-unmapped-type! The Simd type {simd_type} does not have a mapping. Please add one in SimdCheck.cs");
```

Comment:
start with either `!unknown-` or `!wrong-` so it gets sorted along other important/critical logs